pyro/infer/trace_elbo.py

```diff
@@ -49,25 +47,8 @@ def _get_traces(self, model, guide, *args, **kwargs):
         runs the guide and runs the model against the guide with
         the result packaged as a trace generator
         """
-        for i in range(self.num_particles):
-            guide_trace = poutine.trace(guide).get_trace(*args, **kwargs)
-            model_trace = poutine.trace(poutine.replay(model, guide_trace)).get_trace(*args, **kwargs)
-            if infer.is_validation_enabled():
-                check_model_guide_match(model_trace, guide_trace)
-            guide_trace = prune_subsample_sites(guide_trace)
-            model_trace = prune_subsample_sites(model_trace)
-
-            model_trace.compute_log_prob()
-            guide_trace.compute_score_parts()
-            if infer.is_validation_enabled():
-                for site in model_trace.nodes.values():
-                    if site["type"] == "sample":
-                        check_site_shape(site, self.max_iarange_nesting)
-                for site in guide_trace.nodes.values():
-                    if site["type"] == "sample":
-                        check_site_shape(site, self.max_iarange_nesting)
-
-            yield model_trace, guide_trace
+        return iter_importance_traces(num_particles=self.num_particles,
```

Comment:
nit: it might clean up style throughout uses of `iter_importance_trace` if we consistently introduced a temporary `get_traces`:
```py
get_traces = iter_importance_traces(num_particles=self.num_particles, graph_type="flat")
return get_traces(model, guide, *args, **kwargs)
```